### ISSUE TITLE
azurerm_template_deployment crash

### ISSUE THREAD

**Issue by Globegitter** [hashicorp/terraform#13242]
### Terraform Version
Terraform version: 0.9.2 6365269+CHANGES
Go runtime version: go1.8

### Affected Resource(s)
- azurerm_template_deployment

### Terraform Configuration Files
```hcl
resource "azurerm_template_deployment" "test" {
  name                = "${var.cluster_name}"
  resource_group_name = "${var.resource_group_name}"

  template_body = "${file(azuredeploy.json)}"
  parameters = parameters = {
    agentpool1VMSize = "Standard_A2"

    agentpool1VnetSubnetID = "/.../resourceGroups/k8sexample/providers/Microsoft.Network/virtualNetworks/k8sexample_vnet/subnets/k8sexample_vnet_subnet"

    apiServerCertificate = "..."

    apiServerPrivateKey = "..."

    caCertificate = "..."

    clientCertificate = "..."

    clientPrivateKey = "..."

    dockerEngineDownloadRepo = "https://apt.dockerproject.org/repo"

    firstConsecutiveStaticIP = "172.20.10.10"

    kubeConfigCertificate = "..."

    kubeConfigPrivateKey = "..."

    kubernetesAddonManagerSpec = "gcr.io/google_containers/kube-addon-manager-amd64:v6.2"

    kubernetesAddonResizerSpec = "gcr.io/google_containers/addon-resizer:1.6"

    kubernetesDNSMasqSpec = "gcr.io/google_containers/kube-dnsmasq-amd64:1.3"

    kubernetesDashboardSpec = "gcr.io/google_containers/kubernetes-dashboard-amd64:v1.5.1"

    kubernetesExecHealthzSpec = "gcr.io/google_containers/exechealthz-amd64:1.2"

    kubernetesHeapsterSpec = "gcr.io/google_containers/heapster:v1.2.0"

    kubernetesHyperkubeSpec = "gcr.io/google_containers/hyperkube-amd64:v1.5.3"

    kubernetesKubeDNSSpec = "gcr.io/google_containers/kubedns-amd64:1.7"

    kubernetesPodInfraContainerSpec = "gcr.io/google_containers/pause-amd64:3.0"

    linuxAdminUsername = "azureuser"

    location = ""

    masterEndpointDNSNamePrefix = "my-k8s-test"

    masterVMSize = "Standard_A2"

    masterVnetSubnetID = "/subscriptions/...f/resourceGroups/k8sexample/providers/Microsoft.Network/virtualNetworks/k8sexample_vnet/subnets/k8sexample_vnet_subnet"

    networkPolicy = "none"

    servicePrincipalClientId = "..."

    servicePrincipalClientSecret = "..."

    sshRSAPublicKey = "ssh-rsa ..."

    targetEnvironment = "AzurePublicCloud"
  }
  deployment_mode = "Incremental"
  depends_on      = ["null_resource.run_acs_engine"]
}
```

### Panic Output
https://gist.github.com/Globegitter/1811dfecd0f7d76ef351355b2ea9886f

### Expected Behavior
This should have created the resources

### Actual Behavior
It crashed

### Steps to Reproduce
1. Follow the steps at http://danielrhoades.com/2017/03/20/docker-azure-terra/ under `Azure CLI` until the right deployment and parameter files are created with the command ` acs-engine k8s.json`
2. Put the path to the generated `azuredeploy.json` file into the `template_body` and convert the `azuredeploy.parameters.json` into a map of `parameters`
3. `terraform apply`

### References
This is the same deployment as in #13241 just running into separate issues.

**Comment by whiskeyjay** [comment]
I believe this issue is a duplicate of #7353. PR #13670 should stop the crash, adds support for bool and int types for outputs.

**Comment by StephenWeatherford** [comment]
@Globegitter Could you please verify and close (with the current master or when a new version comes out)?  Thanks.

**Comment by tombuildsstuff** [comment]
Hi @Globegitter 

As @StephenWeatherford has said, I believe this has been fixed in #13670 and as such I'm going to close this issue - please reopen if this is still an issue :)

**Comment by Globegitter** [comment]
This seems very reasonable @tombuildsstuff @StephenWeatherford. Will try and test it asap.
